rfc/20241206-oci-registries/3-design-considerations.md

```diff
@@ -0,0 +1,95 @@
+2. **Supporting layers adds complexity**<br />OpenTofu today contains over 300.000 lines of code. Supporting the diff-tar layer format adds complexity to the codebase and increases the resource consumption of the download. Simply downloading a ZIP-blob allows us to reuse much of the code already in place in OpenTofu today.
+
+Therefore, we have decided to use a layout that does not indicate a container image and follows the [ORAS](https://oras.land) conventions with added multi-platform support. See the [Providers](5-providers.md) and [Modules](6-modules.md) documents for details on the respective artifact layouts.
+
+> [!NOTE]
+> In contrast to providers, modules currently have a protocol option. This allows us to integrate them safely without breaking existing tooling. However, to indicate possible OCI layout changes in the future, we will use the protocol prefix of `oci://`.
+
+## Software Bill of Materials (SBOM)
+
+The decision to use a custom image layout as described above profoundly impacts security scanners, which a [large number of our users want to use](2-survey-results.md). We have tested several of the popular security scanners with varying degrees of success.
+
+The popular scanners supporting container images out of the box, such as [Trivy](https://trivy.dev/), only support the default container image-like layout and do not support custom artifact types, such as the one uploaded with [ORAS](https://oras.land/). Other popular tools, such as TFLint, do not support container images at all and need to be configured manually.
+
+However, we have also discovered that many of these tools [already support various SBOM formats](https://trivy.dev/latest/docs/target/sbom/). This gives us the option to let security scanners work on SBOM files instead of scanning a container image and the binary contained therein.
+
+We, therefore, intend to start supporting the publication of SBOM artifacts in the SPDX (JSON) and CycloneDX (JSON and XML) format in the existing OpenTofu Registry by the provider author, and we intend the mirroring tool for OCI registries to mirror these artifacts correctly. Since this file will be included in the `SHA256SUMS` file, the contents of this file will also be signed with the provider GPG key.

```

Comment:
I would like to re-word this and more explicitly move SBOM information out of this RFC.  SBOMs are a nice to have that don't impact the initial code / implementation of this work.  I believe that actually implementing SBOM support should be prioritized and discussed separately.  I'd rather just reference the concept here and say that it is a standard way that we could work around this limitation, instead of locking us into it for implementation.